### PR TITLE
[tlse] internal TLS support for designate

### PR DESCRIPTION
**Pull request by omersch381**
Creates certs for Designate operator when spec.tls.endpoint.internal.enabled: true

### CODE REVIEW

**Comment by omersch381** [comment]
The patch this commit is depended on has probably merged a little too late yesterday, I will try to retest it again.

**Comment by stuggi** [comment]
the functional test errors are real. the designate dependency in the go.mod and api/go.mod need to be first bumped or bumped as part of this PR to have https://github.com/openstack-k8s-operators/designate-operator/commit/ac21d84625baa9eb9a32a30ec08a1dc317b30ef7 .

File: pkg/openstack/designate.go [hunk 1]
```diff
@@ -87,7 +94,7 @@ func ReconcileDesignate(ctx context.Context, instance *corev1beta1.OpenStackCont
 			instance.Spec.Designate.Template.DesignateAPI.Override.Service,
 			instance.Spec.Designate.APIOverride,
 			corev1beta1.OpenStackControlPlaneExposeDesignateReadyCondition,
-			true, // TODO: (mschuppert) disable TLS for now until implemented
+			false, // TODO: (oschwart) could be removed when all integrated service support TLS
 			tls.API{},
```
[the remark this answers is not in the document]
Comment (omersch381) [reply pkg/openstack/designate.go:98]: Yes, I have just compared with other operators and I agree, I will change it, thanks!

**Review by stuggi — APPROVED** [review]
/lgtm

**Pull request merged**